eventq_rabbitmq/lib/eventq_rabbitmq/rabbitmq_eventq_client.rb

```diff
@@ -32,51 +34,83 @@ def register_event(event_type)
       end
 
       def raise_event(event_type, event)
-
         register_event(event_type)
 
-        connection = @client.get_connection
-
         _event_type = EventQ.create_event_type(event_type)
 
-        begin
-        channel = connection.create_channel
+        with_connection do |channel|
+          exchange = @queue_manager.get_exchange(channel, @event_raised_exchange)
 
-        ex = @queue_manager.get_exchange(channel, @event_raised_exchange)
+          message = serialized_message(_event_type, event)
 
-        qm = new_message
-        qm.content = event
-        qm.type = _event_type
+          exchange.publish(message, :routing_key => _event_type)
 
-        if EventQ::Configuration.signature_secret != nil
-          provider = @signature_manager.get_provider(EventQ::Configuration.signature_provider)
-          qm.signature = provider.write(message: qm, secret: EventQ::Configuration.signature_secret)
+          EventQ.logger.debug "[#{self.class}] - Raised event. Message: #{message} | Type: #{event_type}."
         end
+      end
 
-        serialization_provider = @serialization_manager.get_provider(EventQ::Configuration.serialization_provider)
+      def raise_event_in_queue(event_type, event, queue, delay)
+        register_event(event_type)
 
-        message = serialization_provider.serialize(qm)
+        _event_type = EventQ.create_event_type(event_type)
 
-        ex.publish(message, :routing_key => _event_type)
-        rescue => e
+        with_connection do |channel|
+          exchange = @queue_manager.get_exchange(channel, @event_raised_exchange)
 
-          channel.close
-          connection.close
-          raise e
-        end
+          delay_exchange = @queue_manager.get_delay_exchange(channel, queue)
+
+          delay_queue = @queue_manager.create_delay_queue(channel, queue, exchange.name, delay)
+          delay_queue.bind(delay_exchange)
```

Comment:
I would add the routing key to this bind method call:

`delay_queue.bind(delay_exchange, :routing_key => _event_type)`